api/preview/documentdb_types.go

```diff
@@ -68,11 +71,57 @@ type Timeouts struct {
 	StopDelay int32 `json:"stopDelay,omitempty"`
 }
 
+// GatewayTLS defines TLS configuration for the gateway sidecar (Phase 1: certificate provisioning only)
+type GatewayTLS struct {
+	// Mode selects the TLS management strategy.
+	// +kubebuilder:validation:Enum=Disabled;SelfSigned;CertManager;Provided
+	Mode string `json:"mode,omitempty"`
+
+	// CertManager config when Mode=CertManager.
+	CertManager *CertManagerTLS `json:"certManager,omitempty"`
```

Comment:
I would start with `SelfSigned` and `CAProvided` in the beginning unless you have already tested the `CertManager` Option.